resources/src/trade.rs

```diff
@@ -0,0 +1,253 @@
+use chrono::prelude::*;
+use amount::Amount;
+// use asset::AssetIdentifier;
+use offer::PriceRatio;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// A trade represents an offer that was fulfilled between two assets and accounts.
+///
+/// https://www.stellar.org/developers/horizon/reference/resources/trade.html
+#[derive(Debug)]
+pub struct Trade {
+    id: String,
+    offer_id: String,
+    paging_token: String,
+    ledger_close_time: DateTime<Utc>,
+    base_account: String,
+    base_amount: Amount,
+    // base_asset: AssetIdentifier,
+    counter_amount: Amount,
+    counter_account: String,
+    // counter_asset: AssetIdentifier,
+    price: PriceRatio,
+    seller: Seller,
+}
+
+/// The seller involved in the trade.
+#[derive(Eq, PartialEq, Debug, Copy, Clone)]
+pub enum Seller {
+    /// The base created the sell offer.
+    Base,
+    /// The counter created the sell offer.
+    Counter,
+}
+
+impl Seller {
+    /// Returns whether the seller is the base
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Base;
+    /// assert!(seller.is_base());
+    /// ```
+    pub fn is_base(&self) -> bool {
+        *self == Seller::Base
+    }
+
+    /// Returns whether the seller is the counter
+    ///
+    /// ```
+    /// use stellar_resources::TradeSeller;
+    /// let seller = TradeSeller::Counter;
+    /// assert!(seller.is_counter());
+    /// ```
+    pub fn is_counter(&self) -> bool {
+        *self == Seller::Counter
+    }
+}
+
+impl<'de> Deserialize<'de> for Trade {
+    fn deserialize<D>(d: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let rep: TradeRepresentation = TradeRepresentation::deserialize(d)?;
+        let seller = if rep.base_is_seller {
+            Seller::Base
+        } else {
+            Seller::Counter
+        };
+        Ok(Trade {
+            id: rep.id,
+            paging_token: rep.paging_token,
+            ledger_close_time: rep.ledger_close_time,
+            offer_id: rep.offer_id,
+            base_account: rep.base_account,
+            base_amount: rep.base_amount,
+            counter_account: rep.counter_account,
+            counter_amount: rep.counter_amount,
+            price: PriceRatio::from(rep.price),
+            seller: seller,
+        })
+    }
+}
+
+impl Serialize for Trade {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let rep = TradeRepresentation {
+            id: self.id.to_owned(),
+            paging_token: self.paging_token.to_owned(),
+            ledger_close_time: self.ledger_close_time,
+            offer_id: self.offer_id.to_owned(),
+            base_account: self.base_account.to_owned(),
+            base_amount: self.base_amount,
+            base_asset_type: "".to_string(),
+            base_asset_code: None,
+            base_asset_issuer: None,
```

Comment:
Do we do this because all the asset information should be contained in the offer?